src/controls/maplibregl-control.ts

```diff
@@ -0,0 +1,895 @@
+  }
+  get #fitBoundsOptions() {
+    return typeof this.#options.flyTo === "boolean" ? {} : this.#options.flyTo;
+  }
+
+  #flyTo(center: Position, zoom?: number): void {
+    this.#map?.flyTo({ center, ...(zoom ? { zoom } : {}), ...this.#flyToOptions });
+  }
+
+  #fitBounds(bbox: BBox, padding: number, maxZoom?: number): void {
+    this.#map?.fitBounds(
+      [
+        [bbox[0], bbox[1]],
+        [bbox[2], bbox[3]],
+      ],
+      { padding, ...(maxZoom ? { maxZoom } : {}), ...this.#fitBoundsOptions },
+    );
+  }
+
+  #setReverseMarker(coordinates?: Position) {
+    if (this.#options.marker === false || this.#options.marker === null || !this.#map) {
+      return;
+    }
+
+    if (this.#reverseMarker) {
+      if (!coordinates) {
+        this.#reverseMarker.remove();
+
+        this.#reverseMarker = undefined;
+      } else {
+        this.#reverseMarker.setLngLat(coordinates);
+      }
+    } else if (coordinates) {
+      if (this.#options.marker instanceof Function) {
+        this.#reverseMarker = this.#options.marker(this.#map) ?? undefined;
+      } else {
+        this.#reverseMarker = (typeof this.#options.marker === "object" ? new Marker(this.#options.marker) : this.#createMarker())
+          .setLngLat(coordinates)
+          .addTo(this.#map);
+
+        this.#reverseMarker.getElement().classList.add("marker-reverse");
+      }
+    }
+  }
+
+  #setFeatures(markedFeatures: Feature[] | undefined, picked: Feature | undefined): void {
+    if (!this.#map) {
+      return;
+    }
+
+    for (const marker of this.#markers.values()) {
+      marker.remove();
+    }
+
+    this.#markers = new Map();
+
+    this.#setAndSaveData(undefined);
+
+    block: if (picked) {
+      let handled = false;
+
+      if (picked.geometry.type === "GeometryCollection") {
+        const geoms = picked.geometry.geometries.filter((geometry): geometry is Polygon | MultiPolygon => geometry.type === "Polygon" || geometry.type === "MultiPolygon");
+
+        ok: if (geoms.length > 0) {
+          const unioned = union(featureCollection(geoms.map((geom) => feature(geom))));
+
+          if (!unioned) {
+            break ok;
+          }
+
+          setMask(
+            {
+              ...picked,
+              geometry: unioned.geometry,
+            },
+            (data?: GeoJSON) => {
+              this.#setAndSaveData(data);
+            },
+          );
+
+          handled = true;
+        } else {
+          const geometries = picked.geometry.geometries.filter(
+            (geometry): geometry is LineString | MultiLineString => geometry.type === "LineString" || geometry.type === "MultiLineString",
+          );
+
+          if (geometries.length > 0) {
+            this.#setAndSaveData({
+              ...picked,
+              geometry: { type: "GeometryCollection", geometries },
+            });
+
+            handled = true;
+          }
+        }
+      }
+
+      if (handled) {
+        // nothing
```

Comment:
Same here, feels like the logic could be inverted and then the method could return early?